designsafe/apps/workspace/api/views.py

```diff
@@ -780,3 +825,50 @@ def post(self, request, *args, **kwargs):
             },
             encoder=BaseTapisResultSerializer,
         )
+
+
+class AllocationsView(AuthenticatedApiView):
+
+    def _get_allocations(self, username, force=False):
+        """
+        Returns indexed allocation data stored in Django DB, or fetches
+        allocations from TAS and stores them.
+        Parameters
+            ----------
+            username: str
+                TACC username to fetch allocations for.
+            Returns
+            -------
+            dict
+        """
+        user = User.objects.get(username=username)
+        try:
+            if force:
+                logger.info("Forcing TAS allocation retrieval for user:{}".format(username))
+                raise NotFoundError
+            result = {
+                'hosts': {}
+            }
+            result.update(UserAllocations.objects.get(user=user).value)
+            return result
+        except (NotFoundError, UserAllocations.DoesNotExist):
+            # Fall back to getting allocations from TAS
+            allocations = _get_tas_allocations(username)
+            userAllocs = UserAllocations(user=user, value=allocations)
+            userAllocs.save()
```

Comment:
We should also update the allocations cache upon the cached request, otherwise the cache will remain stale and never update. We can kick it off as an async background task so the cached response isn't delayed